lib/cisco_node_utils/bgp.rb

```diff
@@ -360,6 +360,7 @@ def default_enforce_first_as
     # Nvgen as True With optional 'size <size>
     def event_history_cli
       match = config_get('bgp', 'event_history_cli', @get_args)
+      return default_event_history_cli if match == 'size_small'
```

Comment:
Chris and I talked through it and I understand the issue now. I think a better fix would be:

1) Add `auto_default: false` to this property in the YAML:

```diff
 event_history_cli:
+  auto_default: false
   config_get_token_append: '/^(no )?event-history cli(?: size (\S+))?$/'
   config_set_append: '<state> event-history cli <size>'
   default_value: 'size_small'
```

This will cause `config_get` to return `nil` if there is no match instead of returning the string `'size_small'`.

2) Change the getter as follows:

```diff
     # Nvgen as True With optional 'size <size>
     def event_history_cli
       match = config_get('bgp', 'event_history_cli', @get_args)
-      return 'false' if match[0] == 'no '
-      return 'size_' + match[1] if match[1]
+      if match.is_a?(Array)
+        return 'false' if match[0] == 'no '
+        return 'size_' + match[1] if match[1]
+      end
       default_event_history_cli
```